test/new-e2e/tests/agent-platform/install-script/install_script_test.go

```diff
@@ -48,11 +48,20 @@ func TestInstallScript(t *testing.T) {
 	err := json.Unmarshal(platforms.Content, &platformJSON)
 	require.NoErrorf(t, err, "failed to umarshall platform file: %v", err)
 
-	osVersions := strings.Split(*osVersion, ",")
+	osVersions := []string{}
+	// Splitting an empty string results in a slice with a single empty string which wouldn't be useful
+	if *osVersion != "" {
+		osVersions = strings.Split(*osVersion, ",")
+	}
```

Comment:
*suggestion*: 

since we are hardening this part we could add this to make sure that there are no weird encoded spaces

```suggestion
	osVersions := []string{}
	// Splitting an empty string results in a slice with a single empty string which wouldn't be useful
	if strings.TrimFunc(*osVersion, unicode.IsSpace) != "" {
		osVersions = strings.Split(*osVersion, ",")
	}
```

Reply:
In addition from accepting your suggestion I've also made the test fail in an obvious way when no os version is given, rather than pass without running any tests at all. That is in line with the existing behavior, the difference being that now the failure is explicit rather than failing further down for a less obvious reason.